File: chat-service/src/main/java/com/wedding/chat_service/entity/Message.java
package com.wedding.chat_service.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "messages")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Message {

    @Id
    private String id;

    @Indexed
    private Long senderId;

    @Indexed
    private Long receiverId;

    @Indexed
    private Long listingId;

    private String messageContent;

    @Indexed
    private LocalDateTime createdAt = LocalDateTime.now();

    // Optional: Snapshots for display purposes
    private String senderNameSnapshot;
    private String listingTitleSnapshot;
}

File: frontend-app/Frontend/src/app/types/message.ts
export interface SendMessageRequest {
    receiverId: number;
    listingId: number;
    messageContent: string;
    senderNameSnapshot?: string;
    listingTitleSnapshot?: string;
}

export interface VendorMessageResponse {
    id: string;
    senderId: number;
    senderName: string;
    listingId: number;
    listingTitle: string;
    messageContent: string;
    createdAt: string;
}
